### PR TITLE
Add documentation for adaptive partitioning and adaptive task sizing of FTE

### PR DESCRIPTION
**Pull request by linzebing**
## Description

Add documentation for adaptive partitioning and adaptive task sizing of FTE

## Additional context and related issues

Add docs for https://github.com/trinodb/trino/pull/16719 and https://github.com/trinodb/trino/pull/17024

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/admin/fault-tolerant-execution.rst [hunk 1]
```diff
@@ -227,27 +227,17 @@ properties only apply to a ``TASK`` retry policy.
    * - Property name
      - Description
      - Default value
-   * - ``fault-tolerant-execution-target-task-input-size``
```
Comment (jhlodin) [inline docs/src/main/sphinx/admin/fault-tolerant-execution.rst:230]: What's the behavior for users who still have this (and other removed properties) configured?

Comment (linzebing) [reply docs/src/main/sphinx/admin/fault-tolerant-execution.rst:230]: server startup will fail. Let me add this property to `DefunctConfigs`

File: docs/src/main/sphinx/admin/fault-tolerant-execution.rst [hunk 2]
```diff
@@ -258,6 +248,50 @@ properties only apply to a ``TASK`` retry policy.
...
+   * - ``fault-tolerant-execution-arbitrary-distribution-compute-task-target-size-min``
+     - Initial/minimum target input size for non-writer tasks of arbitrary
+       distribution of fault-tolerant execution.
+     - ``512MB``
+   * - ``fault-tolerant-execution-arbitrary-distribution-compute-task-target-size-max``
+     - Maximum target input size for non-writer task of arbitrary distribution
+       of fault-tolerant execution.
+     - ``50GB``
+   * - ``fault-tolerant-execution-arbitrary-distribution-write-task-target-size-growth-period``
+     - The number of tasks we create for given writer stage of arbitrary
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/fault-tolerant-execution.rst:268]: avoid we

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/admin/fault-tolerant-execution.rst [hunk 3]
```diff
@@ -227,27 +227,17 @@ properties only apply to a ``TASK`` retry policy.
...
-       with split weight taken into account. If the weight of splits produced by
-       a catalog denotes that they are lighter or heavier than "standard" split,
-       then the number of splits processed by single task is adjusted
-       accordingly.
-
-       May be overridden for the current session with the
-       ``fault_tolerant_execution_target_task_split_count``
-       :ref:`session property <session-properties-definition>`.
-     - ``64``
+     - ``64MB``
```
[the remark this answers is not in the document]
Comment (linzebing) [reply docs/src/main/sphinx/admin/fault-tolerant-execution.rst:240]: Yeah I know this is done for other places but not for fault-tolerant execution docs. The content of the form is already too crowded. I would recommend a member from docs team to tackle them as a whole.

File: docs/src/main/sphinx/admin/fault-tolerant-execution.rst [hunk 4]
```diff
@@ -258,6 +248,50 @@ properties only apply to a ``TASK`` retry policy.
...
        :ref:`session property <session-properties-definition>`.
      - ``256``
+   * - ``fault-tolerant-execution-arbitrary-distribution-compute-task-target-size-growth-period``
+     - The number of tasks we create for given non-writer stage of arbitrary
+       distribution before we increase task size.
+     - ``64``
+   * - ``fault-tolerant-execution-arbitrary-distribution-compute-task-target-size-growth-factor``
+     - Growth factor for adaptive sizing of non-writer tasks of arbitrary
+       distribution for fault-tolerant execution.
+     - ``1.2``
```
[the remark this answers is not in the document]
Comment (linzebing) [reply docs/src/main/sphinx/admin/fault-tolerant-execution.rst:258]: Lower bound is 1.0. Thanks for the reminder, I have added a commit to enforce this check.

>  think it means you add 20% each time .. right?

Yes. Let me add more color around this one.

File: docs/src/main/sphinx/admin/fault-tolerant-execution.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/admin/fault-tolerant-execution.rst:240]: Okay .. lets me follow up with that work after we get this PR merged.

**Review by losipiuk — APPROVED** [review]

**Review by jhlodin — APPROVED** [review]
Couple more comments, then LGTM

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 5]
```diff
@@ -29,6 +29,16 @@ stages of a query. You can use the following execution policies:
...
   which increases the query wall time overall.
 
+``query.determine-partition-count-for-write-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``false``
+* **Session property:** ``determine_partition_count_for_write_enabled``
+
+Enables determining the number of partitions based on amount of data read and processed by the
```
Comment (colebow) [inline docs/src/main/sphinx/admin/properties-query-management.rst:39]:
```suggestion
Enables determining the number of partitions based on the amount of data read and processed by the
```

**Review by colebow — APPROVED** [review]
Extremely tiny nit, but LGTM.

**Review by mosabua — APPROVED** [review]
This looks good now. Thank you @linzebing .. we will follow up with further improvements as discussed in a separate PR.

File: docs/src/main/sphinx/admin/fault-tolerant-execution.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (jhlodin) [reply docs/src/main/sphinx/admin/fault-tolerant-execution.rst:240]: Followed up with https://github.com/trinodb/trino/pull/17218

**Pull request merged**